File: web/src/modules/reports/models/report.ts
// Report shapes returned by the backend /reports endpoints. The backend owns
// the category taxonomy and all aggregation, so the frontend only renders.

export interface CategoryBreakdown {
  name: string
  icon: string
  color: string
  budget: number
  spent: number
  remaining: number
  pct: number
}

export interface MonthlySpend {
  month: string // YYYY-MM
  total: number
}

File: server/internal/domain/models/report.go
package models

// CategoryBreakdown is the spend for a single category against its budget,
// computed on the backend so the frontend doesn't have to categorize.
type CategoryBreakdown struct {
	Name      string  `json:"name"`
	Icon      string  `json:"icon"`
	Color     string  `json:"color"`
	Budget    float32 `json:"budget"`
	Spent     float32 `json:"spent"`
	Remaining float32 `json:"remaining"`
	Pct       int     `json:"pct"`
}

// MonthlySpend is the summed spend for a single calendar month (YYYY-MM).
type MonthlySpend struct {
	Month string  `json:"month"`
	Total float32 `json:"total"`
}
